README.md

```diff
@@ -198,20 +198,17 @@ Chat completion is available through the [`create_chat_completion`](https://llam
 
 The high-level API also provides a simple interface for function calling.
 
-Note that the only model that supports full function calling at this time is "functionary".
-The gguf-converted files for this model can be found here: [functionary-7b-v1](https://huggingface.co/abetlen/functionary-7b-v1-GGUF)
+The only set of models that supports full function calling at this time is [functionary](https://github.com/MeetKai/functionary). The various gguf-converted files for this set of models can be found [here](https://huggingface.co/meetkai). Functionary is able to intelligently call functions and also analyze any provided function outputs to generate coherent responses. All v2 models of functionary supports **parallel function calling**.
+
+Note that due to discrepancies between llama.cpp and HuggingFace's tokenizers, it is required to provide the path to the HF tokenizer for functionary. They are already included in the respective HF repositories hosting the gguf files.
```

Comment:
What do you think about using `hf_hub_download` to get the model and to acquire the local file path?

```python
from huggingface_hub import hf_hub_download

# Model repository on the Hugging Face model hub
model_repo = "meetkai/functionary-small-v2.2-GGUF"

# File to download
file_name = "functionary-small-v2.2.f16.gguf"

# Download the file
local_file_path = hf_hub_download(repo_id=model_repo, filename=file_name)
local_file_path
```

Reply:
Thanks for the good suggestion! However, I think it will be difficult to download the model with huggingface_hub as huggingface_hub is not a dependency in this project. I'm not sure if @abetlen is ok with setting this as a dependency?

It's kinda pointless if we set up `hf_hub_download` just for the HF AutoTokenizer since we would have already git cloned the repository to get the model.

Reply:
Good to know. `hf_hub_download` does cache the path so for any user that already has it (via transformers, etc.), it won't redownload it.